### ISSUE TITLE
[data request] Pascal VOC

### ISSUE THREAD

**Issue by rodrigob** [tensorflow/datasets#27]
* Name of dataset: Pascal VOC12 with SDB extension
* URL of dataset: http://host.robots.ox.ac.uk/pascal/VOC/ and http://home.bharathh.info/pubs/codes/SBD/download.html
* License of dataset: ?? (to be investigated)
* Short description of dataset and use case(s): the classic detection and semantic segmentation dataset. Used in many works related to weakly supervised learning (since a bit easier than Coco).

Folks who would also like to see this dataset in `tensorflow/datasets`, please +1/thumbs-up so the developers can know which requests to prioritize.

**Comment by ParthS007** [comment]
Hi @rsepassi, I would like to work on adding this dataset. I have gone though gitter discussions and want to get started with this Project. Can you please help me getting started and assign this issue to me.

Thanks

**Comment by rsepassi** [comment]
Great! Let me know when you've accepted the collaborator invite, and I'll
assign you. The guide for adding a new dataset is here:
https://github.com/tensorflow/datasets/blob/master/docs/add_dataset.md

**Comment by ParthS007** [comment]
@rsepassi I have accepted the invite. I will go through the doc and will ask here if I face any difficulty.

Thanks for the help.

**Comment by ParthS007** [comment]
@rodrigob Can you please provide the exact links of training and validation datasets?

**Comment by amanp592** [comment]
@rodrigob I would love to work on this dataset if anyone not working on it can you assign it to me??

**Comment by ParthS007** [comment]
@amanp592 You can work on it.

**Comment by Conchylicultor** [comment]
@ParthS007 @amanp592 Please note that someone already started working on this: https://github.com/tensorflow/datasets/pull/343

**Comment by jpuigcerver** [comment]
Hi,

I'm the author of #343: JFYI my pull request only adds VOC2007, not VOC2012.

**Comment by Conchylicultor** [comment]
@jpuigcerver oops, my bad, indeed. Sorry about that.

**Comment by rodrigob** [comment]
@ParthS007 Any progress on this front ?

**Comment by AdamHillier** [comment]
I needed VOC2012 for something so added it to my fork of tfds. It is simply @jpuigcerver's VOC2007 code adapted very slightly to point to the right tar files.

I've opened a PR #966 to have it merged.

As it includes only the object detection ground truths (axis aligned bounding boxes) and not the segmentation ground truths (or the SDB extension) it doesn't fully satisfy this dataset request, but it might be useful to some people.

(Note that it includes only the training and validation VOC2012 data, as annotations for the test data are not publicly available.)

**Comment by Conchylicultor** [comment]
`voc/2012` is now available. Closing this

**Comment by rodrigob** [comment]
This ticket was not understood.
https://www.tensorflow.org/datasets/catalog/voc
only has bounding boxes,
while 
Pascal VOC12 with SDB extension
includes semantic labeling annotations (as indicated in the ticket) which are still missing in tfds.
It seems that semantic/instance/panoptic segmentation are missing as a category of image annotations.

**Comment by Conchylicultor** [comment]
@rodrigob thanks for the clarification. Re-opening.

**Comment by AdamHillier** [comment]
> This ticket was not understood.
> https://www.tensorflow.org/datasets/catalog/voc
> only has bounding boxes,
> while
> Pascal VOC12 with SDB extension
> includes semantic labeling annotations (as indicated in the ticket) which are still missing in tfds.
> It seems that semantic/instance/panoptic segmentation are missing as a category of image annotations.

I have PR #1198 open to add VOC2012 segmentation with SBD.

**Comment by bhack** [comment]
For the segmentation take a look at https://github.com/keras-team/keras-cv/pull/943
